File: dongyimai_sellergoods_interface/src/main/java/com/sellergoods/service/BrandService.java
package com.sellergoods.service;

import com.dongyimai.bean.TbBrand;
import com.dongyimai.result.PageResult;

import java.util.List;

public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询品牌列表
     * @param pageNum   当前页码
     * @param pageSize  每页查询记录数
     * @return
     */
    public PageResult findPage(Integer pageNum, Integer pageSize,TbBrand brand);

    /**
     * 添加品牌
     * 如果返回影响行不为1则代表出现了错误，抛异常
     * @param brand
     */
    public void addBrand (TbBrand brand) throws Exception;

    /**
     * 根据id查询品牌
     * @param id
     * @return
     */
    public TbBrand findByID (Long id);

    /**
     * 更新品牌
     * @param brand
     * @throws Exception
     */
    public void updateBrand (TbBrand brand)throws Exception;

    /**
     * 根据id删除brand
     * @param id
     */
    public void deleteById (Long id) throws Exception;
}

File: dongyimai_manager_web/src/main/webapp/js/service/brandService.js
app.service('brandService',function($http){

    this.findPage = function(pageNum,pageSize,searchBrand){
        return $http.post('../brand/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize,searchBrand);
    }
    //添加
    this.add = function(brand){
        return $http.post('../brand/add.do',brand);
    }
    //修改
    this.update = function(brand){
        return $http.post('../brand/update.do',brand);
    }
    //获取指定商品id信息
    this.findOne = function(id){
        return $http.get('../brand/findOne.do?id=' + id);
    }
    //删除
    this.dele = function(ids){
        return $http.get('../brand/delete.do?ids='+ids);
    }
});